### PR TITLE
Fix Put(T[] array, int arrayIndex, int count)

### PR DESCRIPTION
**Pull request by DomasM**
After putting array of the same size as buffer in it, next single value
put throws IndexOutOfRangeException.

### CODE REVIEW

**Comment by cyotek** [comment]
Hello,

Thanks for the bug report and fix! I've looked it over and it seems to be OK except for an odd service reference that has appeared - can you clarify what this is? I copied your .cs code changes into my copy so I could poke around (I was curious why you had to modify SkipWrapBufferTest) and it doesn't seem to be required (the service reference that is, not the test changes :)).

Thanks again!

File: Cyotek.Collections.Generic.CircularBuffer.Tests/Cyotek.Collections.Generic.CircularBuffer.Tests.csproj [hunk 1]
```diff
@@ -57,6 +57,9 @@
   <ItemGroup>
     <None Include="packages.config" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+  </ItemGroup>
```
Comment (cyotek) [inline Cyotek.Collections.Generic.CircularBuffer.Tests/Cyotek.Collections.Generic.CircularBuffer.Tests.csproj:62]: Any idea what this reference is?

**Comment by DomasM** [comment]
Hello, this reference was auto-generated by Visual Studio and has something to do with the fact that project is test project. I think it can be safely deleted.

**Comment by cyotek** [comment]
Thanks again for the contribution! I'll get the NuGet package updated shortly.

**Pull request merged**